### DISCUSSION TITLE
does `bootc update` care if the current state is not an ancestor of the updated state?

### DISCUSSION THREAD

**Discussion by beav** [bootc-dev/bootc#251] [Q&A] [answered]
Let's say I have images X and Y which are totally different. I push image X to `my-registry:5000/my-image`. I then make a small addition to X, creating X' which is an OCI image with a new layer on top of X.

I then have a system that's pulling from `my-registry:5000/my-image:latest`. Will bootc work any differently from a user's perspective if I push either X' or Y to `my-image`? My current understanding is that a `bootc update` is like a `git reset --hard` and won't error out if the new image doesn't have the current state as an ancestor. Is this accurate?

**Comment by cgwalters** [comment, marked as answer]
Correct, the prior state does not matter.  There's not a difference between `bootc update` and `bootc switch` (which can change to an arbitrary other image).
